modules/gtksourceview/hm.nix

```
@@ -0,0 +1,23 @@
+{ mkTarget, ... }:
+
+let
+  style = {
+    template = ./template.xml.mustache;
+    extension = ".xml";
+  };
+in
+mkTarget {
+  name = "gtksourceview";
+  humanName = "GTKSourceView";
+
+  configElements =
+    { colors, ... }:
+    {
+      xdg.dataFile = {
+        "gtksourceview-2.0/styles/stylix.xml".source = colors style;
+        "gtksourceview-3.0/styles/stylix.xml".source = colors style;
+        "gtksourceview-4/styles/stylix.xml".source = colors style;
+        "gtksourceview-5/styles/stylix.xml".source = colors style;
+      };
```

What about generating the files from a list of supported versions:

```suggestion
mkTarget {
  name = "gtksourceview";
  humanName = "GTKSourceView";

  configElements =
    { colors, ... }:
    {
      xdg.dataFile = builtins.listToAttrs (
        map
          (
            version:
            lib.nameValuePair "gtksourceview-${version}/styles/stylix.xml" {
              source = colors {
                template = ./template.xml.mustache;
                extension = ".xml";
              };

            }
          )
          [
            "2.0"
            "3.0"
            "4"
            "5"
          ]
      );
```


This makes it easier to add or remove versions in the future. The same pattern is already used in the Blender module:

https://github.com/nix-community/stylix/blob/82a1f36f80a0bdc9875c8b868ea6fe930b09887c/modules/blender/hm.nix#L28-L47

Do not forget to add `lib` into the namespace:

```diff
-{ mkTarget, ... }:
+{ mkTarget, lib, ... }:
```

---

I personally think that the added complexity and decreased readability is not worth it here. We can change it in the future in case we actually need to add new versions.